### ISSUE TITLE
Split IFunction to IFunction and IAggregateFunction 

### ISSUE THREAD

**Issue by sundy-li** [datafuselabs/datafuse#480]
**Summary**
Split  **IFunction** to **IFunction** and **IAggregateFunction**.

The function `eval` in trait `IFunction` makes no sense to IAggregateFunction.

And `accumulate` and `merge`  etc only make sense to IAggregateFunction.

So it's better to split them into different kinds of traits.

- [x] Throw exception for aggregator eval
- [x] Actions Dag

**Comment by BohuTANG** [comment]
If spilt, how to deal this case:
```
select sum(number+2) from numbers_mt(100)
```
Since Binary function `number+2` does not have `accumulate` and `merge`.

**Comment by sundy-li** [comment]
Related #488 

Should we disable `eval` function (throw logic error exceptions) in all aggregator functions?

**Comment by BohuTANG** [comment]
1. ActionsDag is really cool if we replace recursive eval.
2. Throw exception for aggregator `eval` make sense.

**Comment by sundy-li** [comment]
Another case, 

```
mysql>  SELECT 1 > sum(number) from numbers(10);
ERROR 1105 (HY000): Code: 2, displayText = Function Error: 'AggregatorSumFunction' eval unimplemented.
```

**Comment by sundy-li** [comment]
It has been resolved after #550 
```
mysql>  SELECT 1 > sum(number) from numbers(10);
+-------------------+
| (1 > sum(number)) |
+-------------------+
|             false |
+-------------------+
1 row in set (0.01 sec)
```
